fe/fe-core/src/main/java/com/starrocks/catalog/JDBCTable.java

```diff
@@ -60,22 +63,52 @@ public JDBCTable(long id, String name, List<Column> schema, Map<String, String>
         validate(properties);
     }
 
-    public JDBCTable(long id, String name, List<Column> schema, String dbName, String catalogName,
-                     Map<String, String> properties) throws DdlException {
+    public JDBCTable(long id, String name, List<Column> schema, String dbName,
+                     String catalogName, Map<String, String> properties) throws DdlException {
         super(id, name, TableType.JDBC, schema);
+        this.catalogName = catalogName;
         this.dbName = dbName;
+        validate(properties);
+    }
+
+    public JDBCTable(long id, String name, List<Column> schema, List<Column> partitionColumns, String dbName,
+                     String catalogName, Map<String, String> properties) throws DdlException {
+        super(id, name, TableType.JDBC, schema);
         this.catalogName = catalogName;
+        this.dbName = dbName;
+        this.partitionColumns = partitionColumns;
         validate(properties);
     }
 
     public String getResourceName() {
         return resourceName;
     }
 
+    public String getCatalogName() {
+        return catalogName;
+    }
+
+    public String getDbName() {
+        return dbName;
+    }
+
     public String getJdbcTable() {
         return jdbcTable;
     }
 
+    public List<Column> getPartitionColumns() {
+        return partitionColumns;
+    }
+
+    @Override
+    public boolean isUnPartitioned() {
+        if (partitionColumns.size() > 0) {
```

Comment:
may be null?
return partitionColumns == null || partitionColumns.size() == 0;

Reply:
done